include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -1051,6 +1054,54 @@ struct ArenaCfg : Base<OrtArenaCfg> {
 // Custom OPs (only needed to implement custom OPs)
 //
 
+/// <summary>
+/// This struct wraps a raw pointer const OrtNodeArg* that is obtained from
+/// a KernelInfo object. Use it to access the name or type info of an input
+/// or output.
+/// </summary>
+struct NodeArg {
+  explicit NodeArg(const OrtNodeArg* node_arg);
+
+  // Returns a pair consisting of a Ort::AllocatedStringPtr and the string's length.
+  // Ex: auto name = node_arg.GetName(allocator);
+  // name.first.get(); // Get char*
+  // name.second; // Get length
+  std::pair<AllocatedStringPtr, size_t> GetName(OrtAllocator* allocator) const;
+
+  // Returns the node's type and shape information.
+  [[nodiscard]] TypeInfo GetTypeInfo() const;
+
+  constexpr explicit operator const OrtNodeArg*() const noexcept { return p_; }
```

Comment:
>explicit [](http://example.com/codeflow?start=12&length=8)

We do not want that. If we did, the practice is to use get()

Reply:
That's what I thought too, but I did it this way in preparation for using the new `detail::Base<>` class in your PR, which has `constexpr operator contained_type*() const noexcept { return p_; }`. I wanted to make sure that this class was used in the same way to ease merging.

But for now, I can change it to `get()` since it's really not a big deal to change it later.